webui/react/src/hooks/useModal/UserSettings/useModalAddToGroup.tsx

```diff
@@ -0,0 +1,106 @@
+import { Form, message, Select } from 'antd';
+import { FormInstance } from 'antd/lib/form/hooks/useForm';
+import React, { useCallback } from 'react';
+
+import useFeature from 'hooks/useFeature';
+import { updateGroup } from 'services/api';
+import { V1GroupSearchResult } from 'services/api-ts-sdk';
+import Spinner from 'shared/components/Spinner';
+import useModal, { ModalHooks as Hooks } from 'shared/hooks/useModal/useModal';
+import { ErrorType } from 'shared/utils/error';
+import { DetailedUser } from 'types';
+import handleError from 'utils/error';
+
+const FIELD_NAME = 'groups';
+
+interface Props {
+  form: FormInstance;
+  groups: V1GroupSearchResult[];
+}
+
+const ModalForm: React.FC<Props> = ({ form, groups }) => {
+  const rbacEnabled = useFeature().isOn('rbac');
+
+  if (!rbacEnabled) {
+    return null;
+  }
+
+  return (
+    <Spinner spinning={!groups}>
+      <Form form={form} labelCol={{ span: 24 }}>
+        <Form.Item name={FIELD_NAME}>
+          <Select
+            mode="multiple"
+            optionFilterProp="children"
+            placeholder="Select Groups"
+            showSearch>
+            {groups.map((u) => (
```

Comment:
Should filter on group, adding user to it's own group throws an error

Reply:
ended up addressing this by fetching the user's groups